### ISSUE TITLE
Integration version is incremented during publishing yet deployment contains old version

### ISSUE THREAD

**Issue by phantomjinx** [syndesisio/syndesis#3416]
## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Documentation issue or request
</code></pre>

## The problem

1. Integration is created with a version of 1;
2. Integration is published:
  - A copy of the integration is nested in the deployment;
  - The draft integration version is incremented.

Equality comparison of the draft integration and the deployment's integration (_deployment.getSpec()_) is no longer possible since the versions are now different.

Should the version be incremented as a result of publishing rather than editing?

**Comment by zregvart** [comment]
> Should the version be incremented as a result of publishing rather than editing?

Yeah, and we probably should not need the draft until the user elects to _Save as draft_.

**Comment by phantomjinx** [comment]
@zregvart 
Sorry, could you clarify.

Just to be specific:
- I reference the term "draft integration" to mean the integrations that is fetched from the database through the server rest api and held locally as a js object, ie. the items listed on the integrations tab of the UI. Therefore, not sure if we are not confusing terms.
- The incrementing occurs on the "draft integration" (maybe local integration would be better?) after copying itself to the spec reference of the deployment. Thus, the local integration and deployment already differ by version number even though they are identical in all other properties. I don't understand why that would be helpful.

**Comment by zregvart** [comment]
I refer to "draft integration" as one that the client has "saved as draft", i.e. the one that was not (yet) published. Incrementing the version without any user interaction doesn't make sense to me.
